packages/tests/server/streaming.test.ts

```diff
@@ -0,0 +1,213 @@
+import { routerToServerAndClientNew } from './___testHelpers';
+import { TRPCLink, httpBatchLink } from '@trpc/client';
+import { initTRPC } from '@trpc/server';
+import { observable } from '@trpc/server/observable';
+import { konn } from 'konn';
+import superjson from 'superjson';
+import { z } from 'zod';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const unidici = require('undici');
+
+globalThis.fetch = unidici.fetch;
+
+describe('no transformer', () => {
+  const ctx = konn()
+    .beforeEach(() => {
+      const t = initTRPC.create({});
+
+      const router = t.router({
+        deferred: t.procedure
+          .input(
+            z.object({
+              wait: z.number(),
+            }),
+          )
+          .query(async (opts) => {
+            await new Promise<void>((resolve) =>
+              setTimeout(resolve, opts.input.wait),
+            );
+            return opts.input.wait;
+          }),
+      });
+
+      const results: {
+        type: 'next' | 'error';
+        value: unknown;
+        opId: number;
+      }[] = [];
+
+      const linkSpy: TRPCLink<typeof router> = () => {
+        // here we just got initialized in the app - this happens once per app
+        // useful for storing cache for instance
+        return ({ next, op }) => {
+          // this is when passing the result to the next link
+          // each link needs to return an observable which propagates results
+          return observable((observer) => {
+            const unsubscribe = next(op).subscribe({
+              next(value) {
+                console.log('got value', value.result);
+                results.push({
+                  type: 'next',
+                  value,
+                  opId: op.id,
+                });
+                observer.next(value);
```

Comment:
So I'm really not sure how the link chain of observables works... Sorry! 

However from what I've seen in the rest of the codebase, I think it's NOT like an onion where each link peels its own metadata and passes the unwrapped data to the next link, but links are supposed to pass the data as is:

```js
// example from www/docs/client/links/index.md
next(value) {
  console.log('we received value', value);
  observer.next(value);
},
```


So I "fixed" the tests by changing the tests themselves...
```diff
- observer.next((value as any).result);
+ observer.next(value);
```